src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -489,10 +490,6 @@ private async Task ClearRowsAsync(bool acquireLock)
         private async Task ReleaseLeasesAsync(SqlConnection connection, CancellationToken token)
         {
             TelemetryInstance.TrackEvent(TelemetryEventName.ReleaseLeasesStart, this._telemetryProps);
-            this._logger.LogDebugWithThreadId("BEGIN WaitRowsLock - ReleaseLeases");
-            // Don't want to change the "_rows" while another thread is attempting to renew leases on them.
-            await this._rowsLock.WaitAsync(token);
-            this._logger.LogDebugWithThreadId("END WaitRowsLock - ReleaseLeases");
             long newLastSyncVersion = this.RecomputeLastSyncVersion();
```

Comment:
No need to wait for semaphore here. While the `_state` is set to `ProcessingChanges`, the semaphore should be acquired if and only if either of `_rows`, `_state` or `_leaseRenewalLoop` needs to be modified in change-consumption loop and none of the shared class fields are getting modified. Once the `_state` equals `ProcessingChanges`, it is only `ClearRowsAsync` method that is in charge of modifying the rows, hence it is the only method that should be waiting on semaphore.